src/parser/parser_visitor/_expressions.py

```diff
@@ -0,0 +1,106 @@
+from src.expression.constant_value_expression import ConstantValueExpression
+from src.expression.logical_expression import LogicalExpression
+
+
+from src.parser.evaql.evaql_parser import evaql_parser
+
+
+##################################################################
+# EXPRESSIONS
+##################################################################
+class Expressions(evaql_parserVisitor):
+    def visitStringLiteral(self, ctx: evaql_parser.StringLiteralContext):
+        # Fix a bug here; 'VAN' Literal gets converted to "'VAN'";
+        # Multiple quotes should be removed
+
+        if ctx.STRING_LITERAL() is not None:
+            return ConstantValueExpression(ctx.getText()[1:-1])
+        # todo handle other types
+        return self.visitChildren(ctx)
+
+    def visitConstant(self, ctx: evaql_parser.ConstantContext):
+        if ctx.REAL_LITERAL() is not None:
+            return ConstantValueExpression(float(ctx.getText()))
+
+        if ctx.decimalLiteral() is not None:
+            return ConstantValueExpression(self.visit(ctx.decimalLiteral()))
+        return self.visitChildren(ctx)
+
+    def visitLogicalExpression(
+            self, ctx: evaql_parser.LogicalExpressionContext):
+        if len(ctx.children) < 3:
+            # error scenario, should have 3 children
+            return None
+        left = self.visit(ctx.getChild(0))
+        op = self.visit(ctx.getChild(1))
+        right = self.visit(ctx.getChild(2))
+        return LogicalExpression(op, left, right)
+
+    def visitBinaryComparisonPredicate(
+            self, ctx: evaql_parser.BinaryComparisonPredicateContext):
+        left = self.visit(ctx.left)
+        right = self.visit(ctx.right)
+        op = self.visit(ctx.comparisonOperator())
+        return ComparisonExpression(op, left, right)
+
+    def visitNestedExpressionAtom(
+            self, ctx: evaql_parser.NestedExpressionAtomContext):
+        # ToDo Can there be >1 expression in this case
+        expr = ctx.expression(0)
+        return self.visit(expr)
+
+    def visitComparisonOperator(
+            self, ctx: evaql_parser.ComparisonOperatorContext):
+        op = ctx.getText()
+        if op == '=':
+            return ExpressionType.COMPARE_EQUAL
+        elif op == '<':
+            return ExpressionType.COMPARE_LESSER
+        elif op == '>':
+            return ExpressionType.COMPARE_GREATER
+        elif op == '>=':
+            return ExpressionType.COMPARE_GEQ
+        elif op == '<=':
+            return ExpressionType.COMPARE_LEQ
+        elif op == '!=':
+            return ExpressionType.COMPARE_NEQ
+        else:
+            return ExpressionType.INVALID
+
+    def visitLogicalOperator(self, ctx: evaql_parser.LogicalOperatorContext):
+        op = ctx.getText()
+
+        if op == 'OR':
+            return ExpressionType.LOGICAL_OR
+        elif op == 'AND':
+            return ExpressionType.LOGICAL_AND
+        else:
```

Comment:
I see we are not supporting NOT. I was thinking we have NOT supported, because https://github.com/georgia-tech-db/eva/blob/master/src/expression/logical_expression.py

Reply:
We are not yet. We will add when we need be.